internal/objects/upload/handler.go

```diff
@@ -0,0 +1,163 @@
+package upload
+
+import (
+	"context"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/theopenlane/iam/auth"
+
+	"github.com/theopenlane/core/internal/objects"
+	"github.com/theopenlane/core/internal/objects/store"
+	"github.com/theopenlane/core/pkg/metrics"
+	pkgobjects "github.com/theopenlane/core/pkg/objects"
+	"github.com/theopenlane/core/pkg/objects/storage"
+)
+
+// HandleUploads persists metadata, uploads files to storage, and enriches the request context with uploaded file details.
+func HandleUploads(ctx context.Context, svc *objects.Service, files []pkgobjects.File) (context.Context, []pkgobjects.File, error) {
+	if len(files) == 0 {
+		return ctx, nil, nil
+	}
+
+	var uploadedFiles []pkgobjects.File
+
+	for _, file := range files {
+		pkgobjects.AddUpload()
+		metrics.StartFileUpload()
+		startTime := time.Now()
+
+		finish := func(status string) {
+			metrics.FinishFileUpload(status, time.Since(startTime).Seconds())
+			pkgobjects.DoneUpload()
+		}
+
+		orgID, _ := auth.GetOrganizationIDFromContext(ctx)
```

Comment:
We shouldn't ignore the error here 
```suggestion
		orgID, err := auth.GetOrganizationIDFromContext(ctx)
		if err != nil {
		     return ctx, nil, err
		} 
```